mathesar_ui/src/sections/table-view/Header.svelte

```diff
@@ -149,11 +180,43 @@
       </Dropdown>
     </div>
   {/each}
+
   <div class="cell" style="width:{70 + DEFAULT_ROW_RIGHT_PADDING}px;left:{
-      columnPosition.get('__row').width + paddingLeft
-    }px;">
-    <div class="add">
-      +
-    </div>
+    columnPosition.get('__row').width + paddingLeft
+  }px;">
+    <Dropdown closeOnInnerClick={false}
+              contentClass="content"
+              bind:isOpen={newColumnDropdownIsOpen}
+              triggerAppearance="plain">
+
+      <svelte:fragment slot="trigger">
+        <span class="name">Add Column</span>
+      </svelte:fragment>
+
+      <svelte:fragment slot="content">
+        <div class="add-column" style="width:{(70 + DEFAULT_ROW_RIGHT_PADDING) * 1.5}px">
+          <div class="grid">
+
+            <TextInput bind:value={newColumnName}>
+              <svelte:fragment slot="prepend">Name:</svelte:fragment>
+            </TextInput>
+            <Button appearance="primary" disabled={newColumnName.length === 0 || getColumnNames().indexOf(newColumnName) >= 0} on:click={() => addColumn()}>
+              Add
+            </Button>
+
+          </div>
+
+          {#if getColumnNames().indexOf(newColumnName) >= 0}
```

Comment:
As specified in the previous comments, this statement can also use the same computed property.

```suggestion
          {#if isDuplicateColumn}
```

Reply:
I have implemented this as suggested.
Thanks!

Reply:
https://github.com/pegnott/mathesar/blob/754414d202a29033bd437f2d845269d5885a5831/mathesar_ui/src/sections/table-view/Header.svelte#L245